app/helpers/automated_tests_client_helper.rb

```diff
@@ -196,7 +196,7 @@ def self.get_test_server_user
   # Verify the user has the permission to run the tests - admins
   # always have the permission, while student has to
   # belong to the group, and have at least one token.
-  def self.check_user_permission(user, grouping)
+  def self.check_user_permission(user, grouping, assignment)

```

Comment:
You shouldn't need to pass in the assignment; there's an association from `grouping`.